libc/sock/connect-nt.c

```diff
@@ -140,6 +142,14 @@ static textwindows int sys_connect_nt_impl(struct Fd *f, const void *addr,
 
 textwindows int sys_connect_nt(struct Fd *f, const void *addr,
                                uint32_t addrsize) {
+  struct sockaddr_un *sun, nt_sun;
+  if (f->family == AF_UNIX && ((struct sockaddr *)addr)->sa_family == AF_UNIX &&
+      addrsize >= sizeof(struct sockaddr_un)) {
+    sun = (struct sockaddr_un *)addr;
+    nt_sun.sun_family = AF_UNIX;
+    if (__mkwin32_sun_path(sun->sun_path, nt_sun.sun_path) == -1) return -1;
+    addr = &nt_sun;
+  }
```

Comment:
Rather than duplicating this `sockaddr_un` conversion code between `bind-nt.c` and here, it would be better to add `__convert_nt_to_sockaddr_un` and `__convert_sockaddr_un_to_nt` functions similar to the  [`__convert_bsd_to_sockaddr` and `__convert_sockaddr_to_bsd` functions](https://github.com/jart/cosmopolitan/blob/6e6fc38935054db0534d5af4fb99c6193305b946/libc/sock/sockaddr.c#L43-L62), but specialized to  `sockaddr_un`.